deploy/docker/hook_manager.py

```
@@ -0,0 +1,512 @@
+            
+            for name in allowed_builtins:
+                if hasattr(builtins, name):
+                    safe_builtins[name] = getattr(builtins, name)
+            
+            namespace = {
+                '__name__': f'user_hook_{hook_point}',
+                '__builtins__': safe_builtins
+            }
+            
+            # Add commonly needed imports
+            exec("import asyncio", namespace)
```

Ruff (0.13.1)

Use of `exec` detected

(S102)